app/models/satisfaction_survey_feature_metrics.rb

```diff
@@ -0,0 +1,54 @@
+class SatisfactionSurveyFeatureMetrics
+  include ActionView::Helpers::NumberHelper
+
+  def response_rate(
+    start_time,
+    end_time = Time.zone.now.end_of_day
+  )
+    success_count = application_forms_with_feedback(start_time, end_time).count.to_f
+    fail_count = application_forms_with_no_feedback(start_time, end_time).count.to_f
+    return nil if (fail_count + success_count).zero?
+
+    success_count / (fail_count + success_count)
+  end
+
+  def formatted_response_rate(
+    start_time,
+    end_time = Time.zone.now.end_of_day
+  )
+    format_as_percentage(response_rate(start_time, end_time))
+  end
+
+private
+
+  def format_as_percentage(ratio)
+    return 'n/a' if ratio.nil?
+
+    percentage = number_with_precision(
+      100.0 * ratio,
+      precision: 1,
+      strip_insignificant_zeros: true,
+    )
+    "#{percentage}%"
+  end
+
+  def application_forms_with_feedback(start_time, end_time)
+    ApplicationForm
+      .where(
+        recruitment_cycle_year: RecruitmentCycle.current_year,
+      )
+      .where('application_forms.submitted_at BETWEEN ? AND ?', start_time, end_time)
+      .where.not(feedback_satisfaction_level: nil).where.not(feedback_suggestions: nil)
```

Comment:
```suggestion
      .where.not(feedback_satisfaction_level: '').where.not(feedback_suggestions: '')
```
There are quite a few records in the prod DB with empty string values for these fields. `.where.not(feedback_suggestions: '')` will filter those out as well as `null` values (because `null` isn't equal to or not equal to any value).

Reply:
Ah okay 👍, that's my fault for forgetting about null!